src/pages/ReferenceGuide.tsx

```diff
@@ -13,13 +18,31 @@ export interface ReferenceGuideProperties extends CompileRemoteBlockOptions {
 }
 
 export default class ReferenceGuide extends WidgetBase<ReferenceGuideProperties> {
+	private _renderSupplementalPage() {
+		const { repo, branch, path, page } = this.properties;
+		const pages: any = this.meta(Block).run(referenceGuideBlock)({
+			repo,
+			branch,
+			path: `${path}/supplemental.md`,
+			locale: getLanguageFromLocale(i18n.locale)
+		});
+
+		if (pages && pages[page]) {
+			return <Page hasLeftSideMenu>{pages[page]}</Page>;
+		}
+	}
+
 	protected render() {
 		const { route, repo, branch, path, page } = this.properties;
 
 		return (
-			<Section>
-				<ReferenceGuideMenu route={route} repo={repo} branch={branch} path={path} page={page} />
-				<RemotePage repo={repo} branch={branch} path={path} relativeUrl={page} hasLeftSideMenu />
+			<Section key="section">
+				<ReferenceGuideMenu route={route} repo={repo} branch={branch} path={path} />
+				{page === 'introduction' || page === 'basic-usage' ? (
+					<RemotePage repo={repo} branch={branch} path={`${path}/${page}.md`} hasLeftSideMenu />
+				) : (
+					this._renderSupplementalPage()
```

Comment:
It looks like `_renderSupplemental` makes a call to `reference-guide.block`, which in turn calls out to `compile-remote.block` and ultimately returns a `Page`. This sounds similar to what `RemotePage` already does internally. Is it possible to augment the existing `RemotePage` component so it can support supplemental page fetching via a prop, instead of manually fetching content into a `Page`?

Reply:
Done